### ISSUE TITLE
Is Ubuntu 16.04 supported by HWI 2.0+?

### ISSUE THREAD

**Issue by kiminuo** [bitcoin-core/HWI#506]
Hi,

when `hwi` (2.0.1) is run on Ubuntu 16.04 it reports:

```bash
$ ./hwi
[11765] Error loading Python lib '/tmp/_MEImoo770/libpython3.6m.so.1.0': dlopen: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.25' not found (required by /tmp/_MEImoo770/libpython3.6m.so.1.0)
```

I'm not sure whether HWI 2.0+ is supposed to be run on newer Ubuntu versions or whether this is a bug.

Thanks!

PS: Ubuntu 18.04 reports:

```
[12221] Error loading Python lib '/tmp/_MEIJQ9F0E/libpython3.6m.so.1.0': dlopen: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /tmp/_MEIJQ9F0E/libpython3.6m.so.1.0)
```

**Comment by achow101** [comment]
Do you see this issue with 2.0.0 or earlier?

If not, I think the cause was #500. Python built in a newer version probably requires a newer glibc that is not present on older versions of Ubuntu.

**Comment by kiminuo** [comment]
> Do you see this issue with 2.0.0 or earlier?

I tested it right now and I can see the error on 2.0.1 but NOT on 2.0.0 on my Ubuntu 18.04.

**Comment by molnard** [comment]
@achow101 can this be fixed or are you planning to? Otherwise, older Linux support has to be dropped. Similarly, you did with macOS, like building the binaries with older dependencies.

**Comment by achow101** [comment]
@molnard I'll look into building with an older Linux for the next release. However HWI is currently low priority for me so I have not had the time to investigate.

**Comment by kiminuo** [comment]
> If not, I think the cause was #500. Python built in a newer version probably requires a newer glibc that is not present on older versions of Ubuntu.

I think so too.

* https://packages.debian.org/source/stretch/glibc mentions `2.24`
* https://packages.debian.org/source/buster/glibc mentions `2.28`

My Ubuntu 18.04 reports `glibc` version `Version: 2.27-3ubuntu1.4` (using `sudo apt-cache show libc6`).

**Comment by Szpoti** [comment]
Under Ubuntu 16.04.7 I get
``` Error loading Python lib '/tmp/_MEI38U91B/libpython3.6m.so.1.0': dlopen: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.25' not found (required by /tmp/_MEI38U91B/libpython3.6m.so.1.0) ``` 
when I try to run HWI as well.

**Comment by molnard** [comment]
The fix working on Ubuntu 18.04. However, it seems like on Ubuntu 16.04 there is another issue https://github.com/zkSNACKs/WalletWasabi/pull/5861#issuecomment-851432331. We did not elaborate on that much as Ubuntu 16.04 is not anymore supported with Wasabi for this reason https://github.com/zkSNACKs/WalletWasabi/pull/5867. Some minor tests remaining that will be done tomorrow, will report back with the final result https://github.com/zkSNACKs/WalletWasabi/pull/5861

**Comment by prusnak** [comment]
> it seems like on Ubuntu 16.04 there is another issue [zkSNACKs/WalletWasabi#5861 (comment)](https://github.com/zkSNACKs/WalletWasabi/pull/5861#issuecomment-851432331).

It seems you forgot to install udev rules on Ubuntu 16.04 - https://github.com/zkSNACKs/WalletWasabi/pull/5861#issuecomment-851702243
